### ISSUE TITLE
Support arbitary long sequences

### ISSUE THREAD

**Issue by bratao** [facebookresearch/pytext#628]
Hello Team, thank you for this awesome work!

For some tagger tasks we have an arbitrary long sequence size. And right now Pytext requires to set a fixed size beforehand for each sequence.

This size cannot be too big, because of vanishing gradients problem( for RNN) and the non convergence of LSTM.

https://machinelearningmastery.com/handle-long-sequences-long-short-term-memory-recurrent-neural-networks/ provides some nice solutions for this problem.

My observation about the viability to PyText, I see those two options as more promising:

**Truncated Backpropagation Through Time**

**Stateful LSTM**

But I have questions for example how this would fit with a CRF layer

There is any way to train a tagger with VERY long ( like 10000+ tokens) sequences in Pytext?

Thank you!

**Comment by borguz** [comment]
I would suggest just chunking you input into fixed-sized lengths and training on that.  As you mention, it's difficult for RNN's to models very long context.  So I doubt you'll lose anything by doing this.  
That being said, the BiLSTM representation module does support passing in state:  https://github.com/facebookresearch/pytext/blob/f35987dc7c9c4e5f26aad9bb2ac1ae398063c788/pytext/models/representations/bilstm.py#L75
We used to use this for language modeling in the past, but removed the functionality because it did not improve results.
